client/state/site-settings/exporter/actions.js

```diff
@@ -83,28 +96,32 @@ export function advancedSettingsFail( siteId, error ) {
  * @param  {Number}   siteId  The ID of the site to export
  * @return {Function}         Action thunk
  */
-export function startExport( siteId ) {
-	return ( dispatch ) => {
+export function startExport( siteId, exportAll = true ) {
+	return ( dispatch, getState ) => {
 		if ( ! siteId ) {
 			return;
 		}
 
 		dispatch( {
 			type: EXPORT_START_REQUEST,
-			siteId: siteId
+			siteId,
+			exportAll,
 		} );
 
+		const state = getState();
+		const advancedSettings = exportAll ? null : prepareExportRequest( state, siteId );
+
 		const success =
 			() => dispatch( exportStarted( siteId ) );
 
 		const failure =
 			error => dispatch( exportFailed( siteId, error ) );
 
 		return wpcom.undocumented()
-			.startExport( siteId )
+			.startExport( siteId, advancedSettings )
 			.then( success )
```

Comment:
Interesting discussion on mapping, I wish I'd come across that earlier while thinking about this 😄 

I had considered mapping with a previous iteration of the export endpoints, taking inspiration from `fromAPI` as used in the importer. There was previously a lot of overhead involved in converting to the API data, for example:

```js
{ post_type: 'post', start_date: '2016-06', end_date: '2016-06' }
```
had to become:
```js
{ post_type: 'post', post_start_date: '2016-06-01', post_end_date: '2016-06-30' }
```

In the end, I opted to do this on the server and change the endpoints - especially as they're not yet in use, we can alter them as needed. With the changes introduced in r135999-wpcom, the data coming from/to the API no longer needs any modifications, other than the snake_case/camelCase issue.

Do you think we should be switching snake_case <=> camelCase here?